### DISCUSSION TITLE
Get custom response data from custom model server comand

### DISCUSSION THREAD

**Discussion by mishaltripathi** [eclipse-emfcloud/emfcloud#195] [ModelServer] [answered]
Hello,

I am working on an application which uses EMFCloud as server and theia on frontend. I need to implement custom command which should return json data as part of response. Now the problem is that modelserverapi edit() method returns boolean response but here we need to return custom data which comes from command operation.

Can anyone have an idea on how to achieve this or any other alternate way using model server api.

Thanks you!

**Comment by ndoschek** [comment, marked as answer]
Hi @mishaltripathi,

to directly return a Json object via the edit API, you would have to extend the API you are using (e.g. the `ModelServerClientApiV2`) and also extend the ModelServer itself by adding a separate endpoint to return the Json object response.

There is one example available via the `emfcloud-modelserver` example server and the `emfcloud-modelserver-theia` example application. It triggers a custom 'counter' command that returns the current counter object in Json format.
You can test that by starting up the example application and use the menu `Custom` to trigger this custom command.
See also this short clip:

https://user-images.githubusercontent.com/47414588/215058202-28ac9699-3bf1-43ba-a462-2c11151d8b67.mp4

Some useful starting points are:
- https://github.com/eclipse-emfcloud/emfcloud-modelserver/blob/master/examples/org.eclipse.emfcloud.modelserver.example/src/main/java/org/eclipse/emfcloud/modelserver/example/CoffeeRouting.java
- https://github.com/eclipse-emfcloud/emfcloud-modelserver-theia/blob/97b0990564698c437b700e51f16524d4893417fd/examples/dev-example/src/node/model-server-client.ts#L51

HTH and best regards,
Nina

**Reply by mishaltripathi** [reply]
Sorry for late response. Thanks a lot @ndoschek!! I will check this.
